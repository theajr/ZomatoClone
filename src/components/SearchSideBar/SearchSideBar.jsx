/* eslint-disable react/jsx-wrap-multilines */
import { Divider } from '@material-ui/core';
import Checkbox from '@material-ui/core/Checkbox';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormGroup from '@material-ui/core/FormGroup';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormLabel from '@material-ui/core/FormLabel';
import { makeStyles } from '@material-ui/core/styles';
import React, { useEffect } from 'react';
import Loader from '../Loader';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  formControl: {},
}));

const SearchSideBar = ({
  categories = [],
  selectedCuisines = [],
  fetchCategories,
  toggleSearchItem,
  searchOptions,
  selectedCategories = [],
  cuisines = [],
  loading,
  search,
}) => {
  console.log('dsf', selectedCuisines);
  const classes = useStyles();

  useEffect(() => {
    if (!categories.length) fetchCategories();
    return () => {};
  }, []);

  useEffect(() => {
    search({
      ...searchOptions,
      category: selectedCategories.join(','),
    });
    return () => {};
  }, [selectedCategories]);

  const handleChange = event => {
    toggleSearchItem('categories', event.target.name);
  };

  const handleChangeCuisine = event => {
    toggleSearchItem('cuisines', event.target.name);
  };
  if (loading) return <Loader />;
  return (
    <div>
      <FormControl component="fieldset" className={classes.formControl}>
        <FormLabel component="legend">Categories</FormLabel>
        <FormGroup>
          {categories.map(c => {
            return (
              <FormControlLabel
                key={c.id}
                control={
                  <Checkbox
                    onChange={handleChange}
                    checked={selectedCategories.includes(`${c.id}`)}
                    name={`${c.id}`}
                  />
                }
                label={c.name}
              />
            );
          })}
        </FormGroup>
      </FormControl>
      <Divider style={{ margin: 10, borderTop: `2px solid teal` }} />
      <FormControl component="fieldset" className={classes.formControl}>
        <FormLabel component="legend">Cuisines</FormLabel>
        <FormGroup>
          {cuisines.map(c => {
            return (
              <FormControlLabel
                key={c}
                control={
                  <Checkbox
                    onChange={handleChangeCuisine}
                    checked={selectedCuisines.includes(c)}
                    name={c}
                  />
                }
                label={c}
              />
            );
          })}
        </FormGroup>
      </FormControl>
    </div>
  );
};

SearchSideBar.propTypes = {};

export default SearchSideBar;
