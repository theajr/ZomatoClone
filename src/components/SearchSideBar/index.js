import { connect } from 'react-redux';
import SearchSideBar from './SearchSideBar';
import { fetchCategories } from '../../actions/categories';
import { toggleSearchOption } from '../../actions/sortSearchFilterOptions';
import { searchRestaurants } from '../../actions/restarant';

const mapStateToProps = ({
  restaurant,
  categories,
  location,
  sortSearchFilterOptions,
}) => {
  const allCuisines = restaurant.restaurants.map(r => r.cuisines);
  const cuisines = Array.from(new Set(allCuisines.flat(2)));

  return {
    cuisines,
    categories: categories.categories,
    loading: categories.loading,
    searchOptions: {
      ...sortSearchFilterOptions.search,
      cityId: location.selectedLocation.id,
    },
    selectedCategories: sortSearchFilterOptions.search.categories,
    selectedCuisines: sortSearchFilterOptions.search.cuisines,
  };
};
const mapDispatchToProps = dispatch => {
  return {
    fetchCategories: () => dispatch(fetchCategories()),
    toggleSearchItem: (key, value) => dispatch(toggleSearchOption(key, value)),
    search: (q, cityId) => {
      dispatch(searchRestaurants(q, cityId));
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchSideBar);
