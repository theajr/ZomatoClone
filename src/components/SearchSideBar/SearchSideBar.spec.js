import React from 'react';
import { shallow } from 'enzyme';
import SearchSideBar from './SearchSideBar';

describe('<SearchSideBar />', () => {
  test('renders', () => {
    const wrapper = shallow(<SearchSideBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
