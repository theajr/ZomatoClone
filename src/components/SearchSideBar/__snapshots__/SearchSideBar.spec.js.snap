// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchSideBar /> renders 1`] = `ShallowWrapper {}`;
